2007-02-08  Arnaud Charlet  <charlet@adacore.com>

	* pixman/src/pixman.h: Fix build under Solaris x86

2006-10-27  Arnaud Charlet  <charlet@adacore.com>

	* src/cairo-png.c: Work around compilation error under AIX

2006-02-27  Arnaud Charlet  <charlet@adacore.com>

	* src/cairoint.h, pixman/src/slim_internal.h: Fix build failure on
	Solaris with GCC 3.x

--- src/cairoint.h.old	Mon Feb 27 14:27:41 2006
+++ src/cairoint.h	Mon Feb 27 14:45:52 2006
@@ -94,7 +94,7 @@
 #endif
 
 /* slim_internal.h */
-#if (__GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 3)) && defined(__ELF__)
+#if (__GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 3)) && defined(__ELF__) && !defined (sun)
 #define cairo_private		__attribute__((__visibility__("hidden")))
 #else
 #define cairo_private
--- pixman/src/slim_internal.h.old	Mon Feb 27 14:29:28 2006
+++ pixman/src/slim_internal.h	Mon Feb 27 14:46:26 2006
@@ -47,7 +47,7 @@
 /* ??? Not marked with "slim" because that makes it look too much
    like the function name instead of just an attribute.  */
 
-#if (__GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 3)) && defined(__ELF__)
+#if (__GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 3)) && defined(__ELF__) && !defined (sun)
 #define pixman_private	__attribute__((__visibility__("hidden")))
 #else
 #define pixman_private
--- src/cairo-png.c.orig	Wed Nov  1 03:27:38 2006
+++ src/cairo-png.c	Wed Jan 10 18:50:34 2007
@@ -37,6 +37,11 @@
 
 #include <errno.h>
 #include "cairoint.h"
+
+#ifdef _AIX
+#undef jmpbuf
+#endif
+
 #include <png.h>
 
 /* Unpremultiplies data and converts native endian ARGB => RGBA bytes */
--- pixman/src/pixman.h.old	2007-02-08 04:57:38.713993000 -0500
+++ pixman/src/pixman.h	2007-02-08 05:17:09.618879000 -0500
@@ -100,7 +100,13 @@ SOFTWARE.
 #include "pixman-remap.h"
 
 #if (__GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 3)) && defined(__ELF__)
+
+#if defined (sun) && defined (__i386__)
+#define pixman_private
+#else
 #define pixman_private		__attribute__((__visibility__("hidden")))
+#endif
+
 #elif defined(__SUNPRO_C) && (__SUNPRO_C >= 0x550)
 #define pixman_private		__hidden
 #else /* not gcc >= 3.3 and not Sun Studio >= 8 */
